### PR TITLE
Kingdom Hearts: Minor Logic Fixes

### PR DESCRIPTION
**Pull request by gaithern**
Please format your title with what portion of the project this pull request is
targeting and what it's changing.

ex. "MyGame4: implement new game" or "Docs: add new guide for customizing MyGame3"

## What is this fixing or adding?
Adjusts a few logic rules for some missed scenarios that may cause unreachable (or at least difficult-to-reach) locations.  These issues were brought to my attention by Goblin God.

## How was this tested?
Goblin God and I ran a recent 100% seed to confirm the changes were being taken into account.

### CODE REVIEW

File: worlds/kh1/Rules.py [hunk 1]
```diff
@@ -377,6 +386,7 @@ def set_rules(kh1world):
     add_rule(kh1world.get_location("Monstro Chamber 6 Other Platform Chest"),
         lambda state: (
             state.has("High Jump", player)
+            and state.has("Progressive Glide", player)
```
Comment (Exempt-Medic) [inline worlds/kh1/Rules.py:389]:
```suggestion
            state.has_all({
                "High Jump",
                "Progressive Glide"}, player)
```

File: worlds/kh1/Rules.py [hunk 2]
```diff
@@ -387,6 +397,7 @@ def set_rules(kh1world):
     add_rule(kh1world.get_location("Monstro Chamber 6 Raised Area Near Chamber 1 Entrance Chest"),
         lambda state: (
             state.has("High Jump", player)
+            and state.has("Progressive Glide", player)
```
Comment (Exempt-Medic) [inline worlds/kh1/Rules.py:400]:
```suggestion
            state.has_all({
                "High Jump",
                "Progressive Glide"}, player)
```

**Review by ScipioWright — APPROVED** [review]
LGTM, didn't actually confirm if the logic is correct, but for minor things like this I can reasonably assume the dev figured out what's correct already.

**Review by Exempt-Medic — CHANGES_REQUESTED** [review]
Two small comments

**Review by Exempt-Medic — APPROVED** [review]
LGTM. Tested plando'ing a Blizzard onto "Agrabah Defeat Kurt Zisa Ansem's Report 11" which was valid before the changes and lead to inaccessible locations after them.

**Review by NewSoupVi — APPROVED** [review]
This seems important enough to merge still

**Pull request merged**